build_runner/test/generate/build_integration_test.dart

```diff
@@ -79,6 +79,37 @@ main(List<String> args) async {
           ])
         ]).validate();
       });
+
+      test('--output creates a merged directory under the provided root',
+          () async {
+        // Run a build and validate the full rebuild output.
+        var result = await runDart('a', 'tool/build.dart',
+            args: ['build', '--output', 'build:web']);
+        expect(result.exitCode, 0, reason: result.stderr as String);
+        await d.dir('a', [
+          d.dir('build', [
+            d.dir('web', [d.file('a.txt.copy', 'a')])
```

Comment:
we _don't_ want a directory named `web` under this. You have implemented part of this, which is filtering, but we also need to change the structure of the output.

Reply:
Fixed. Added a corresponding test.